### PR TITLE
Create type-safe React Redux store with TypeScript

### PR DESCRIPTION
**Pull request by jkoutavas**
Moved-over to using redux/typescript scheme as defined here: https://joshuaavalon.io/create-type-safe-react-redux-store-with-typescript

There was a tsc error that needs to be resolved after applying the article's implementation:

```
$ yarn tsc
yarn run v1.22.4
$ /Users/jay/dev/repos/heynow/typescript-redux-react-native-thunk/myTSReduxApp/node_modules/.bin/tsc
store/provider.tsx:11:7 - error TS2322: Type 'Store<State, Action<ActionTypes> & Readonly<{ payload: Employee[]; }>> & { dispatch: unknown; }' is not assignable to type 'Store'.
  Type 'Store<State, Action<ActionTypes> & Readonly<{ payload: Employee[]; }>> & { dispatch: unknown; }' is not assignable to type 'Store<State, Action<string>>'.
    Types of property 'dispatch' are incompatible.
      Type 'Dispatch<Action<ActionTypes> & Readonly<{ payload: Employee[]; }>>' is not assignable to type 'Dispatch<Action<string>>'.
        Type 'Action<string>' is not assignable to type 'Action<ActionTypes> & Readonly<{ payload: Employee[]; }>'.
          Type 'Action<string>' is not assignable to type 'Action<ActionTypes>'.
            Type 'string' is not assignable to type 'ActionTypes'.

11 const store: Store = createStore(
         ~~~~~


Found 1 error.
```

With the help of @vhesener, this PR got sorted-out and we seem to have a good solid basis for a Redux implementation. Thank you @joshuaavalon for the inspiration.

### CODE REVIEW

File: myTSReduxApp/App.tsx [hunk 1]
```diff
@@ -56,22 +59,3 @@ const styles = StyleSheet.create({
...
-});
-
-type StateProps = ReturnType<typeof mapStateToProps>;
-type DispatchProps = ReturnType<typeof mapDispatchToProps>;
-export default connect<StateProps, DispatchProps, {}, IStateModel>(
-  mapStateToProps,
-  mapDispatchToProps,
-  // TODO: work-out the tslint warning on the App parameter
-  // @ts-ignore
-)(App);
```
Comment (jkoutavas) [inline myTSReduxApp/App.tsx:77]: So happy to see connect() gone!

File: myTSReduxApp/store/pageReducer/hook.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import {useSelector} from 'react-redux';
+
+import {selectPageList} from './select';
+
+export const usePageList = () => useSelector(selectPageList);
```
Comment (jkoutavas) [inline myTSReduxApp/store/pageReducer/hook.ts:5]: I particularly like the idea of providing a `usePageList` hook that in turn encapsulates state using the `selectPageList` selector.

As the article covers (see link in PR description), it's good not to expose the structure of state to its consumers, with a hook+selector, you can hide those details and if you restructure the state later on, you don't impact the hooks' users.

File: myTSReduxApp/store/pageReducer/select.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import {createSelector} from 'reselect';
+
+import {State} from '../type';
+
+import {PageReducerState} from './reducer';
+
+export function selectPageReducerState(state: State): PageReducerState {
+  return state.pageReducer;
+}
```
Comment (jkoutavas) [inline myTSReduxApp/store/pageReducer/select.ts:9]: And here is that selector I was mentioning in my previous comment. Very clean encapsulation.

**Comment by dusty-commits** [comment]
Pushed fixes. Let me know if explanation is needed

**Comment by jkoutavas** [comment]
> Pushed fixes. Let me know if explanation is needed

@vhesener, thank you. Your latest commit makes sense to me. It'll be interesting when there's more than one reducer employed (pun intended). I'm thinking about adding a second one to this example app next, just to look that over.

```
$ yarn tsc
yarn run v1.22.4
$ /Users/jay/dev/repos/heynow/typescript-redux-react-native-thunk/myTSReduxApp/node_modules/.bin/tsc
✨  Done in 4.28s.
```

 I've removed the draft status off of this PR and updated its description. It's open to your approval now.

**Review by dusty-commits — APPROVED** [review]

**Pull request merged**